src/auth/oauth2/core/qgsauthoauth2config.cpp

```diff
@@ -319,29 +320,43 @@ bool QgsAuthOAuth2Config::loadConfigTxt(
   const QByteArray &configtxt, QgsAuthOAuth2Config::ConfigFormat format
 )
 {
-  QByteArray errStr;
-  bool res = false;
+  QString errStr;
 
   switch ( format )
   {
     case JSON:
     {
-      const QVariant variant = QJsonWrapper::parseJson( configtxt, &res, &errStr );
-      if ( !res )
+      const QVariant variant = QgsJsonUtils::parseJson( configtxt.toStdString(), errStr );
+      if ( !errStr.isEmpty() )
       {
         QgsDebugError( QStringLiteral( "Error parsing JSON: %1" ).arg( QString( errStr ) ) );
-        return res;
+        return false;
       }
       const QVariantMap variantMap = variant.toMap();
-      // safety check -- qvariant2qobject asserts if an non-matching property is found in the json
-      for ( QVariantMap::const_iterator iter = variantMap.constBegin(); iter != variantMap.constEnd(); ++iter )
-      {
-        const QVariant property = this->property( iter.key().toLatin1() );
-        if ( !property.isValid() ) // e.g. not a auth config json file
-          return false;
-      }
 
-      QJsonWrapper::qvariant2qobject( variantMap, this );
+      setApiKey( variantMap.value( QStringLiteral( "apiKey" ) ).toString() );
```

Comment:
what happens if the variantMap lacks one of the item ? Wouldn't it be preferable not to call the set method then ?